### ISSUE TITLE
Getting an error loading `active_support/core_ext/hash/transform_values` on Rails 4.1

### ISSUE THREAD

**Issue by jaytennier** [amcaplan/simple_jsonapi_client#7]
I'm trying to use `simple_jsonapi_client` in a Rails 4.1 app but I'm getting the following error when I try to do anything (e.g. run `rails`, `rspec`, `rake`, etc.):

```bash
$ be rails server
/Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/simple_jsonapi_client-0.2.2/lib/simple_jsonapi_client/base.rb:2:in `require': cannot load such file -- active_support/core_ext/hash/transform_values (LoadError)
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/simple_jsonapi_client-0.2.2/lib/simple_jsonapi_client/base.rb:2:in `<top (required)>'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/simple_jsonapi_client-0.2.2/lib/simple_jsonapi_client.rb:4:in `require'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/simple_jsonapi_client-0.2.2/lib/simple_jsonapi_client.rb:4:in `<module:SimpleJSONAPIClient>'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/simple_jsonapi_client-0.2.2/lib/simple_jsonapi_client.rb:3:in `<top (required)>'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:81:in `require'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:81:in `block (2 levels) in require'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `each'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `block in require'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `each'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `require'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler.rb:114:in `require'
	from /Users/jay/Sites/simple_jsonapi_client_test/test_app/config/application.rb:7:in `<top (required)>'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.1.16/lib/rails/commands/commands_tasks.rb:79:in `require'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.1.16/lib/rails/commands/commands_tasks.rb:79:in `block in server'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.1.16/lib/rails/commands/commands_tasks.rb:76:in `tap'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.1.16/lib/rails/commands/commands_tasks.rb:76:in `server'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.1.16/lib/rails/commands/commands_tasks.rb:40:in `run_command!'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.1.16/lib/rails/commands.rb:17:in `<top (required)>'
	from /Users/jay/Sites/simple_jsonapi_client_test/test_app/bin/rails:9:in `require'
	from /Users/jay/Sites/simple_jsonapi_client_test/test_app/bin/rails:9:in `<top (required)>'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/lib/spring/client/rails.rb:28:in `load'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/lib/spring/client/rails.rb:28:in `call'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/lib/spring/client/command.rb:7:in `call'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/lib/spring/client.rb:30:in `run'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/bin/spring:49:in `<top (required)>'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/lib/spring/binstub.rb:11:in `load'
	from /Users/jay/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/spring-1.7.2/lib/spring/binstub.rb:11:in `<top (required)>'
	from /Users/jay/Sites/simple_jsonapi_client_test/test_app/bin/spring:13:in `require'
	from /Users/jay/Sites/simple_jsonapi_client_test/test_app/bin/spring:13:in `<top (required)>'
	from bin/rails:3:in `load'
	from bin/rails:3:in `<main>'
```

From what I can see it's trying to load `active_support/core_ext/hash/transform_values` which wasn't [introduced until Rails 4.2](https://github.com/rails/rails/commit/b2cf8b251aac39c1e3ce71bc1de34a2ce5ef52b1).

The simplest way to reproduce this is to create a brand new Rails 4.1 app, add `simple_jsonapi_client` to the `Gemfile` and then try to run `rails server`. You should get the 
error above.

**Comment by amcaplan** [comment]
Oof, sorry this made your day more difficult. And thanks for looking into why this was happening.

Honestly, it seems like we use `transform_values` in just one place as a convenience. Requiring `active_support` at all is just for that plus `symbolize_keys`. Neither of those things is so complicated we couldn't just write some helper methods to do the same stuff, and then we could shave off the ActiveSupport dependency entirely.

I'll try to deal with this in the next few days.

**Comment by amcaplan** [comment]
Hey @jaytennier, can you have a look at https://github.com/amcaplan/simple_jsonapi_client/pull/9 and verify it works for you? Maybe require in your app like:

```ruby
gem 'simple_jsonapi_client', git: 'https://github.com/amcaplan/simple_jsonapi_client.git', branch: 'remove-activesupport-dependency'
```

**Comment by jaytennier** [comment]
Hey @amcaplan, thanks for taking the time to work on this! I gave the new branch a try and I'm no longer getting the original error but if I try to fetch a collection I now get the following:

```ruby
[16] pry(main)> ResourceClass.fetch_all(connection: connection).entries
NoMethodError: undefined method `dig' for #<Hash:0x000000052ec908>
from /var/www/project/shared/bundle/ruby/2.2.0/bundler/gems/simple_jsonapi_client-31f7d0a4ff84/lib/simple_jsonapi_client/redirection/fetch_all.rb:33:in `block (2 levels) in internal_enumerator'
```

As you can see from the error message the project is using Ruby 2.2 and unfortunately `Hash#dig` wasn't introduced until Ruby 2.3. The good news is that if I extend Hash to include a `dig` method (I used the implementation from [the `ruby_dig` gem](https://github.com/Invoca/ruby_dig/blob/master/lib/ruby_dig.rb)) everything works as expected!

So as far as this issue is concerned it looks like you've got it fixed. I'm not sure if you want to add support for Ruby 2.2 or not (I totally understand if you don't) but if not maybe you can just add something to the README that explains the situation and points users at the `ruby_dig` gem to help fix the problem.

Thanks again for all your help!

**Comment by amcaplan** [comment]
I realized today that I never actually committed the change to `master`, nor did I handle the `dig` issue. 🤦‍♂  I hope it hasn't caused you tons of trouble, @jaytennier...

Anyway, it's all merged in, and I cut a new version, 0.2.4, freshly released on Rubygems.

Have a great day!
